ludwig/schema/decoders/utils.py

```diff
@@ -0,0 +1,82 @@
+from dataclasses import field
+
+from marshmallow import fields, ValidationError
+
+from ludwig.constants import TYPE
+from ludwig.decoders.registry import get_decoder_classes, get_decoder_cls
+from ludwig.schema import utils as schema_utils
+
+
+def get_decoder_conds(feature_type: str):
+    """Returns a JSON schema of conditionals to validate against decoder types for specific feature types."""
+    conds = []
+    for decoder in get_decoder_classes(feature_type):
+        decoder_cls = get_decoder_cls(feature_type, decoder).get_schema_cls()
+        other_props = schema_utils.unload_jsonschema_from_marshmallow_class(decoder_cls)["properties"]
+        other_props.pop("type")
+        decoder_cond = schema_utils.create_cond(
+            {"type": decoder},
+            other_props,
+        )
+        conds.append(decoder_cond)
+    return conds
+
+
+def DecoderDataclassField(feature_type: str, default: str):
+    """Custom dataclass field that when used inside a dataclass will allow the user to specify a decoder config.
+
+    Returns: Initialized dataclass field that converts an untyped dict with params to a decoder config.
+    """
+
+    class DecoderMarshmallowField(fields.Field):
+        """Custom marshmallow field that deserializes a dict for a valid decoder config from the decoder_registry
+        and creates a corresponding `oneOf` JSON schema for external usage."""
+
+        def _deserialize(self, value, attr, data, **kwargs):
+            if value is None:
+                return None
+            if isinstance(value, dict):
+                if TYPE in value and value[TYPE] in get_decoder_classes(feature_type):
+                    enc = get_decoder_cls(feature_type, default).get_schema_cls()
```

Comment:
nit: should this be `dec` for decoder instead of `enc` for encoder?

Reply:
Yes good catch, copy and pasted the same logic so forgot to change it. Changing now!